### ISSUE TITLE
php_fastcgi env variables resulting in unexpected values in phpinfo

### ISSUE THREAD

**Issue by bfinlay** [caddyserver/caddy#3860]
caddy placeholders are not returning expected values in $_SERVER

caddy version: `v2.2.1 h1:Q62GWHMtztnvyRU+KPOpw6fNfeCD3SkwH7SfT1Tgt2c=`

some particular examples:
* `hostport` returns the same value as host
* `port` returns no value

example caddyfile.
note that the connection to php-fpm is by socket in this configuration

```
localhost:80, 127.0.0.1:80, localhost:443, 127.0.0.1:443

root * {$PWD}/public
php_fastcgi * unix/{$HOME}/.config/php-fpm.sock {
    env test_dir {dir}
    env test_file {file}
    env test_host {host}
    env test_hostport {hostport}
    env test_port {port}
    env test_method {method}
    env test_path {path}
    env test_query {query}
    env test_remote {remote}
    env test_remote_host {remote_host}
    env test_remote_port {remote_port}
    env test_scheme {scheme}
    env test_uri {uri}
    env test_tls_cipher {tls_cipher}
    env test_tls_version {tls_version}
    env test_tls_client_fingerprint {tls_client_fingerprint}
    env test_tls_client_issuer {tls_client_issuer}
    env test_tls_client_serial {tls_client_serial}
    env test_tls_client_subject {tls_client_subject}
}
file_server
```

result from $_SERVER variable for `http://localhost/server?a=apple&b=ball`:
```
  'test_dir' => '',
  'test_file' => 'index.php',
  'test_host' => 'localhost',
  'test_hostport' => 'localhost',
  'test_method' => 'GET',
  'test_path' => 'index.php',
  'test_port' => '',
  'test_query' => 'a=apple&b=ball',
  'test_remote' => '[::1]:49283',
  'test_remote_host' => '::1',
  'test_remote_port' => '49283',
  'test_scheme' => 'http',
  'test_tls_cipher' => '',
  'test_tls_client_fingerprint' => '',
  'test_tls_client_issuer' => '',
  'test_tls_client_serial' => '',
  'test_tls_client_subject' => '',
  'test_tls_version' => '',
  'test_uri' => 'index.php?a=apple&b=ball',
```

and for `https://localhost/server?a=apple&b=ball`:
```
  'test_dir' => '',
  'test_file' => 'index.php',
  'test_host' => 'localhost',
  'test_hostport' => 'localhost',
  'test_method' => 'GET',
  'test_path' => 'index.php',
  'test_port' => '',
  'test_query' => 'a=apple&b=ball',
  'test_remote' => '[::1]:60038',
  'test_remote_host' => '::1',
  'test_remote_port' => '60038',
  'test_scheme' => 'https',
  'test_tls_cipher' => 'TLS_AES_128_GCM_SHA256',
  'test_tls_client_fingerprint' => '',
  'test_tls_client_issuer' => '',
  'test_tls_client_serial' => '',
  'test_tls_client_subject' => '',
  'test_tls_version' => 'tls1.3',
  'test_uri' => 'index.php?a=apple&b=ball',
```

**Comment by mholt** [comment]
Are those two variables the only ones that are confusing?

I think they're working as intended. According to [the docs](https://caddyserver.com/docs/json/apps/http/#docs):

> {http.request.hostport} | The host and port from the request's Host header
>
> {http.request.port} | The port part of the request's Host header

So what is the `Host` header? It probably just doesn't have a port, which is pretty normal for servers on the standard ports.

**Comment by bfinlay** [comment]
I see.   Thank you for the quick reply.

The documentation page that I found was this one: https://caddyserver.com/docs/caddyfile/concepts#placeholders and then followed that through to this one: https://caddyserver.com/docs/conventions#placeholders but I did not follow through to the "HTTP app sets placeholders" link on that page, so I did not see see the "Host header" clarification before.

The Host header is "localhost" so that corresponds to the behavior that you describe.

Is there a way to get an equivalent to nginx [$server_port](https://nginx.org/en/docs/http/ngx_http_core_module.html#var_server_port)?    Is there another documentation page with more placeholders?
